rasa/nlu/extractors/mitie_entity_extractor.py

```diff
@@ -133,50 +187,105 @@ def _prepare_mitie_sample(training_example: Message) -> Any:
+
+        Args:
+            message: a user message
+            mitie_model: MitieModel containing a `mitie.total_word_feature_extractor`
 
-        ents = self.extract_entities(
-            message.get(TEXT), message.get(TOKENS_NAMES[TEXT]), mitie_feature_extractor
+        Returns:
+            a list of dictionaries describing the entities
+        """
+        text = message.get(TEXT)
+        tokens = message.get(TOKENS_NAMES[TEXT])
+
+        entities = []
+        token_texts = [token.text for token in tokens]
+        mitie_entities = self._ner.extract_entities(
+            token_texts, mitie_model.word_feature_extractor,
         )
-        extracted = self.add_extractor_name(ents)
-        message.set(ENTITIES, message.get(ENTITIES, []) + extracted, add_to_output=True)
+        for e in mitie_entities:
+            if len(e[0]):
+                start = tokens[e[0][0]].start
+                end = tokens[e[0][-1]].end
+
+                entities.append(
+                    {
+                        ENTITY_ATTRIBUTE_TYPE: e[1],
+                        ENTITY_ATTRIBUTE_VALUE: text[start:end],
+                        ENTITY_ATTRIBUTE_START: start,
+                        ENTITY_ATTRIBUTE_END: end,
+                        ENTITY_ATTRIBUTE_CONFIDENCE: None,
+                    }
+                )
+        return entities
 
     @classmethod
     def load(
         cls,
-        meta: Dict[Text, Any],
-        model_dir: Text,
-        model_metadata: Metadata = None,
-        cached_component: Optional["MitieEntityExtractor"] = None,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
         **kwargs: Any,
-    ) -> "MitieEntityExtractor":
+    ) -> MitieEntityExtractorGraphComponent:
         """Loads trained component (see parent class for full docstring)."""
         import mitie
 
-        file_name = meta.get("file")
-
-        if not file_name:
-            return cls(meta)
-
-        classifier_file = os.path.join(model_dir, file_name)
-        if os.path.exists(classifier_file):
-            extractor = mitie.named_entity_extractor(classifier_file)
-            return cls(meta, extractor)
-        else:
-            return cls(meta)
-
-    def persist(self, file_name: Text, model_dir: Text) -> Optional[Dict[Text, Any]]:
+        try:
+            with model_storage.write_to(resource) as model_path:
+                ner_file = model_path / cls.MITIE_RESOURCE_FILE
+                if not ner_file.exists():
+                    raise FileNotFoundError(
+                        f"Expected a MITIE extractor file at {ner_file}."
+                    )
+                ner = mitie.named_entity_extractor(str(ner_file))
+                return MitieEntityExtractorGraphComponent(
+                    config, model_storage, resource, ner=ner
+                )
+        except FileNotFoundError:
```

Comment:
Should we catch `ValueError` too?

Reply:
the old implementation doesn't - and mitie.py actually raises a generic `Exception` ... I've just added an except that catches that mitie exception

Reply:
`model_storage.read_from` throws it if the resource doesn't exist yet, and we've been handling it in all the other `load`s.

Reply:
ah ok, wasn't aware of that 👍